src/UnityExtension/Assets/Editor/GitHub.Unity/UI/BranchesView.cs

```diff
@@ -33,42 +33,34 @@ class BranchesView : Subview
         private const string DeleteBranchButton = "Delete";
         private const string CancelButtonLabel = "Cancel";
 
-        private bool showLocalBranches = true;
-        private bool showRemoteBranches = true;
-
         [NonSerialized] private int listID = -1;
-        [NonSerialized] private BranchTreeNode newNodeSelection;
         [NonSerialized] private BranchesMode targetMode;
 
-        [SerializeField] private BranchTreeNode activeBranchNode;
-        [SerializeField] private BranchTreeNode localRoot;
+        [SerializeField] private Tree treeLocals = new Tree();
+        [SerializeField] private Tree treeRemotes = new Tree();
         [SerializeField] private BranchesMode mode = BranchesMode.Default;
         [SerializeField] private string newBranchName;
-        [SerializeField] private List<Remote> remotes = new List<Remote>();
         [SerializeField] private Vector2 scroll;
-        [SerializeField] private BranchTreeNode selectedNode;
+        [SerializeField] private bool disableDelete;
 
         [SerializeField] private CacheUpdateEvent lastLocalAndRemoteBranchListChangedEvent;
         [NonSerialized] private bool localAndRemoteBranchListHasUpdate;
 
-        [SerializeField] private GitBranch[] localBranches;
-        [SerializeField] private GitBranch[] remoteBranches;
+        [SerializeField] private List<GitBranch> localBranches;
+        [SerializeField] private List<GitBranch> remoteBranches;
 
         public override void InitializeView(IView parent)
         {
             base.InitializeView(parent);
             targetMode = mode;
         }
 
+
         public override void OnEnable()
         {
             base.OnEnable();
             AttachHandlers(Repository);
-
-            if (Repository != null)
-            {
-                Repository.CheckLocalAndRemoteBranchListChangedEvent(lastLocalAndRemoteBranchListChangedEvent);
-            }
+            Repository.CheckLocalAndRemoteBranchListChangedEvent(lastLocalAndRemoteBranchListChangedEvent);
```

Comment:
Repository is not longer possibly null then?

Reply:
By the time the `BranchesView` is displayed we always have a `Repository`